src/Core/System/SalesChannel/Command/SalesChannelListCommand.php

```diff
@@ -0,0 +1,129 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\System\SalesChannel\Command;
+
+use Shopware\Core\Framework\Context;
+use Shopware\Core\Framework\DataAbstractionLayer\EntityRepositoryInterface;
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\System\Currency\CurrencyEntity;
+use Shopware\Core\System\Language\LanguageEntity;
+use Shopware\Core\System\SalesChannel\Aggregate\SalesChannelDomain\SalesChannelDomainEntity;
+use Shopware\Core\System\SalesChannel\SalesChannelCollection;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class SalesChannelListCommand extends Command
+{
+    protected static $defaultName = 'sales-channel:list';
+
+    /**
+     * @var EntityRepositoryInterface
+     */
+    private $salesChannelRepository;
+
+    public function __construct(
+        EntityRepositoryInterface $salesChannelRepository
+    ) {
+        $this->salesChannelRepository = $salesChannelRepository;
+
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->addOption(
+            'output',
+            '0',
+            InputOption::VALUE_OPTIONAL,
+            'Output mode',
+            'table'
+        );
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $headers = [
+            'id',
+            'Name',
+            'Active',
+            'Maintenance',
+            'Default Language',
+            'Languages',
+            'Default Currency',
+            'Currencies',
+            'Domains'
+        ];
+
+
+        $criteria = new Criteria();
+        $criteria->addAssociations(['language', 'languages', 'currency', 'currencies', 'domains']);
+        /** @var SalesChannelCollection $salesChannels */
+        $salesChannels = $this->salesChannelRepository->search($criteria, Context::createDefaultContext());
+        foreach ($salesChannels as $salesChannel) {
+            $data[] = [
+                $salesChannel->getId(),
+                $salesChannel->getName(),
+                $salesChannel->getActive() ? '1' : '0',
+                $salesChannel->isMaintenance() ? '1' : '0',
```

Comment:
```suggestion
                $salesChannel->isMaintenance() ? 'on' : 'off',
```
Maybe?

Reply:
@shyim  I wanted to use integers so it could be easily casted to a bool if ever used programatically (like in a build script).